### PR TITLE
Add Email Scope to specify the people who can use the bot.

### PR DESCRIPTION
**Pull request by tsa87**
Works, however deprecates previous macros created without the `Scope` field.

### CODE REVIEW

**Review by miasya — APPROVED** [review]
I like that it uses email scopes - is the next step to add the saving names in added to space?

File: src/main/java/com/google/flourbot/datastorage/FirebaseDataStorage.java [hunk 1]
```diff
@@ -45,8 +45,8 @@ public FirebaseDataStorage() {
     // Create a query to find a macro named macroName belonging to userEmail
     Query query =
         db.collection(COLLECTION_NAME)
-            .whereEqualTo(USER_IDENTIFIER, userEmail)
-            .whereEqualTo(MACRO_IDENTIFIER, macroName);
+            .whereEqualTo(MACRO_IDENTIFIER, macroName)
+            .whereArrayContains(SCOPE_IDENTIFIER, userEmail);
```
Comment (SydneyCaulfeild) [inline src/main/java/com/google/flourbot/datastorage/FirebaseDataStorage.java:49]: I'm not sure if using an index helps or speeds the query up, but worth looking into it (https://firebase.corp.google.com/project/stepladder-2020/database/firestore/indexes).

File: src/main/java/com/google/flourbot/datastorage/FirebaseDataStorage.java [hunk 2]
```diff
@@ -59,7 +59,7 @@ public FirebaseDataStorage() {
       document = querySnapshot.get().getDocuments().get(0);
       return document.exists() ? Optional.of(document) : Optional.empty();
     } catch (IndexOutOfBoundsException e) {
-      throw new IllegalStateException(e);
+      return Optional.empty();
```
Comment (SydneyCaulfeild) [inline src/main/java/com/google/flourbot/datastorage/FirebaseDataStorage.java:62]: Thanks for finding this!!

**Review by SydneyCaulfeild — APPROVED** [review]

File: src/main/java/com/google/flourbot/datastorage/FirebaseDataStorage.java (hunk 2 above)
Comment (lipczak-g) [inline src/main/java/com/google/flourbot/datastorage/FirebaseDataStorage.java:62]: Oh, if IndexOutOfBoundsException is possible, we should check isEmpty first before .get(0) and return empty optional if that's the case. Exceptions are expensive and complicated, if we can we should avoid them.

Comment (tsa87) [reply src/main/java/com/google/flourbot/datastorage/FirebaseDataStorage.java:62]: Oh I see. I don't see a .emtpy() method on `DocumentSnapshot`  type though.

**Pull request merged**